homeassistant/components/litterrobot/switch.py

```diff
@@ -67,20 +70,22 @@ def is_on(self) -> bool | None:
         return bool(getattr(self.robot, self.entity_description.key))
 
     @property
-    def icon(self) -> str | None:
+    def icon(self) -> str:
         """Return the icon."""
         icon_on, icon_off = self.entity_description.icons
         return icon_on if self.is_on else icon_off
 
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the switch on."""
-        if set_fn := self.entity_description.set_fn:
-            await self.perform_action_and_assume_state(set_fn(self.robot), True)
+        await self.perform_action_and_assume_state(
+            self.entity_description.set_fn(self.robot), True
+        )
```

Comment:
Mmmm. Maybe two lines is better than three

```suggestion
        set_fn = self.entity_description.set_fn
        await self.perform_action_and_assume_state(set_fn(self.robot), True)
```